modules/programs/antidote.nix

```
@@ -33,21 +34,29 @@ in {
 
   config = mkIf cfg.enable {
     home.packages = [ cfg.package ];
-
-    home.file."${relToDotDir ".zsh_plugins.txt"}".text = zPluginStr cfg.plugins;
-
-    ### move zsh_plugins.txt
-    programs.zsh.initExtraBeforeCompInit = ''
-      ## home-manager/antidote begin :
-      source ${cfg.package}/share/antidote/antidote.zsh
-      ${optionalString cfg.useFriendlyNames
-      "zstyle ':antidote:bundle' use-friendly-names 'yes'"}
-      bundlefile=$HOME/${relToDotDir ".zsh_plugins.txt"}
-      zstyle ':antidote:bundle' file $bundlefile
-      staticfile=$HOME/${relToDotDir ".zsh_plugins.zsh"}
-      zstyle ':antidote:static' file $staticfile
-      antidote load $bundlefile $staticfile
-      ## home-manager/antidote end
-    '';
+    programs.zsh.initExtraBeforeCompInit = (
+
+      let
+        configFiles = pkgs.runCommand "hm_antidote-files" { } ''
+          mkdir -p $out/cache
+          echo "${zPluginStr cfg.plugins}" > $out/zsh_plugins.txt
+          ##generate hash_id to make staticfile binded with hm
+          echo -n "$(basename $out | cut -d - -f 1)" > $out/hash_id
+        '';
+      in ''
+        ## home-manager/antidote begin :
+        source ${cfg.package}/share/antidote/antidote.zsh
+        ${optionalString cfg.useFriendlyNames
+        "zstyle ':antidote:bundle' use-friendly-names 'yes'"}
+
+        bundlefile=${configFiles}/zsh_plugins.txt;
+        zstyle ':antidote:bundle' file $bundlefile
+        staticfile="/tmp/tmp_hm_zsh_plugins.zsh$(cat ${configFiles}/hash_id)"
```

I'm not very familiar with zsh but I think you can avoid cat here:
```suggestion
        staticfile="/tmp/tmp_hm_zsh_plugins.zsh$(< "${configFiles}/hash_id")"
```

---

I use a function to parse hash id from path.